src/topology/continuous_on.lean

```diff
@@ -1004,3 +1009,22 @@ continuous_snd.continuous_on
 lemma continuous_within_at_snd {s : set (α × β)} {p : α × β} :
   continuous_within_at prod.snd s p :=
 continuous_snd.continuous_within_at
+
+lemma continuous_within_at.fst {f : α → β × γ} {s : set α} {a : α}
```

Comment:
streamline the proof with `continuous_at.comp_continuous_within_at`? Same thing for the next lemma.